File: UCSD/Data-Structures-and-Algorithms-Specialization/Course1-AlgorithmicToolbox/Assignment1/MaximumPairwiseProduct/MaxPairwiseProd.cpp
#include <iostream>
#include <vector>



long long MaxPairwiseProduct(const std::vector<int>& numbers)
{
    int max1, max2;

    max1 = max2 = 0;
    
    for (int i = 0; i < numbers.size(); i++) {
      if (numbers[i] > max1) {
        max2 = max1;
        max1 = numbers[i];
      }
      else if (numbers[i] > max2) max2 = numbers[i];
    }

    return (long long) max1 * max2;
}



int main()
{
    int n;
    std::cin >> n;

    std::vector<int> numbers(n);
    for (int i = 0; i < n; i++) std::cin >> numbers[i];
    
    std::cout << MaxPairwiseProduct(numbers) << std::endl;
}

File: UCSD/Data-Structures-and-Algorithms-Specialization/Course1-AlgorithmicToolbox/Assignment1/MaximumPairwiseProduct/MaxPairwiseProd.py
def max_pairwise_product(numbers):
    max1 = max2 = 0
    for i in range(len(numbers)):
        if numbers[i] > max1:
            max2 = max1
            max1 = numbers[i]
        elif numbers[i] > max2: 
            max2 = numbers[i]

    return max1 * max2



if __name__ == "__main__":
    n = int(input())
    numbers = list(map(int, input().split()))
    print(max_pairwise_product(numbers))
